dot/parachain/availability-store/availability_store.go

```diff
@@ -36,142 +80,431 @@ type AvailabilityStoreSubsystem struct {
+	}
+	err = asb.chunk.Flush()
+	if err != nil {
+		return fmt.Errorf("writing chunk batch: %w", err)
+	}
+	err = asb.meta.Flush()
+	if err != nil {
+		return fmt.Errorf("writing meta batch: %w", err)
+	}
+	err = asb.unfinalized.Flush()
+	if err != nil {
+		return fmt.Errorf("writing unfinalized batch: %w", err)
+	}
+	err = asb.pruneByTime.Flush()
+	if err != nil {
+		return fmt.Errorf("writing prune by time batch: %w", err)
+	}
+	return nil
+}
+
+// Reset resets the batch and returns the error
+func (asb *AvailabilityStoreBatch) Reset() {
+	asb.available.Reset()
+	asb.chunk.Reset()
+	asb.meta.Reset()
+	asb.unfinalized.Reset()
+	asb.pruneByTime.Reset()
 }
 
 // NewAvailabilityStore creates a new instance of AvailabilityStore
 func NewAvailabilityStore(db database.Database) *AvailabilityStore {
 	return &AvailabilityStore{
-		availableTable: database.NewTable(db, avaliableDataPrefix),
-		chunkTable:     database.NewTable(db, chunkPrefix),
-		metaTable:      database.NewTable(db, metaPrefix),
+		available:   database.NewTable(db, availableDataPrefix),
+		chunk:       database.NewTable(db, chunkPrefix),
+		meta:        database.NewTable(db, metaPrefix),
+		unfinalized: database.NewTable(db, unfinalizedPrefix),
+		pruneByTime: database.NewTable(db, pruneByTimePrefix),
 	}
 }
 
 // loadAvailableData loads available data from the availability store
-func (as *AvailabilityStore) loadAvailableData(candidate common.Hash) (*AvailableData, error) {
-	resultBytes, err := as.availableTable.Get(candidate[:])
+func (as *AvailabilityStore) loadAvailableData(candidate parachaintypes.CandidateHash) (*AvailableData, error) {
+	resultBytes, err := as.available.Get(candidate.Value[:])
 	if err != nil {
-		return nil, fmt.Errorf("getting candidate %v from available table: %w", candidate, err)
+		return nil, fmt.Errorf("getting candidate %v from available table: %w", candidate.Value, err)
 	}
 	result := AvailableData{}
-	err = json.Unmarshal(resultBytes, &result)
+	err = scale.Unmarshal(resultBytes, &result)
 	if err != nil {
 		return nil, fmt.Errorf("unmarshalling available data: %w", err)
 	}
 	return &result, nil
 }
 
-// loadMetaData loads metadata from the availability store
-func (as *AvailabilityStore) loadMetaData(candidate common.Hash) (*CandidateMeta, error) {
-	resultBytes, err := as.metaTable.Get(candidate[:])
+// writeAvailableData put available data in the availabilityBatch of the given batch
+func (as *AvailabilityStore) writeAvailableData(batch *AvailabilityStoreBatch,
+	candidate parachaintypes.CandidateHash, data AvailableData) error {
+	dataBytes, err := scale.Marshal(data)
 	if err != nil {
-		return nil, fmt.Errorf("getting candidate %v from available table: %w", candidate, err)
+		return fmt.Errorf("marshalling available data: %w", err)
+	}
+	err = batch.available.Put(candidate.Value[:], dataBytes)
+	if err != nil {
+		return fmt.Errorf("storing available data for candidate %v: %w", candidate, err)
+	}
+	return nil
+}
```

Comment:
right now `as` remains unused. I would suggest to remove batch from the arguments. 
Create a new batch, perform the task and flush the batch in the end.

Edit the comment accordingly